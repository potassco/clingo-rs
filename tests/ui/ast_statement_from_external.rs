use clingo::*;

fn main() {
    let loc = ast::Location::default();
    let sym = Symbol::create_id("test", true).unwrap();
    let term1: ast::Term = ast::symbolic_term(&loc, &sym).unwrap().into();
    let atom1 = ast::symbolic_atom(term1).unwrap();
    let body = [];
    let ext = ast::external(&loc, atom1, &body, ExternalType::Free).unwrap();
    let stmt : ast::Statement = ext.into();

    drop(ext);
    let _end = stmt;
}