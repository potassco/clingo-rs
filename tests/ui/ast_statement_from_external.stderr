error[E0382]: use of moved value: `ext`
  --> tests/ui/ast_statement_from_external.rs:12:10
   |
9  |     let ext = ast::external(&loc, atom1, &body, ExternalType::Free).unwrap();
   |         --- move occurs because `ext` has type `clingo::ast::External<'_>`, which does not implement the `Copy` trait
10 |     let stmt : ast::Statement = ext.into();
   |                                     ------ `ext` moved due to this method call
11 |
12 |     drop(ext);
   |          ^^^ value used here after move
   |
note: `into` takes ownership of the receiver `self`, which moves `ext`
  --> $RUST/core/src/convert/mod.rs
   |
   |     fn into(self) -> T;
   |             ^^^^
help: you can `clone` the value and consume it, but this might not be your desired behavior
   |
10 |     let stmt : ast::Statement = ext.clone().into();
   |                                     ++++++++
